analysis/utility_analysis.py

```diff
@@ -160,3 +163,118 @@ def _create_aggregate_error_compound_combiner(
                     error_quantiles))
     return utility_analysis_combiners.AggregateErrorMetricsCompoundCombiner(
         internal_combiners, return_named_tuple=False)
+
+
+################# New code  ##################
+# perform_utility_analysis_new will be renamed to perform_utility_analysis.
+def perform_utility_analysis_new(
+        col,
+        backend: pipeline_backend.PipelineBackend,
+        options: analysis.UtilityAnalysisOptions,
+        data_extractors: Union[pipeline_dp.DataExtractors,
+                               analysis.PreAggregateExtractors],
+        public_partitions=None,
+        return_per_partition: bool = False):
+    """Performs utility analysis for DP aggregations.
+
+    Args:
+      col: collection where all elements are of the same type.
+      backend: PipelineBackend with which the utility analysis will be run.
+      options: options for utility analysis.
+      data_extractors: functions that extract needed pieces of information
+        from elements of 'col'. In case if the analysis performed on
+        pre-aggregated data, it should have type PreAggregateExtractors
+        otherwise DataExtractors.
+      public_partitions: A collection of partition keys that will be present
+        in the result. If not provided, the utility analysis with private
+        partition selection will be performed.
+      return_per_partition: if true, it returns tuple, with the 2nd element
+        utility analysis per partitions.
+    Returns:
+         if return_per_partition == False:
+            returns collections which contains metrics.UtilityReport with
+            one report per each input configuration
+        else returns tuple, with the 1st element as in first 'if' clause and the
+        2nd a collection which contains utility analysis results per partition).
```

Comment:
What exactly does utility analysis results per partition mean exactly?

Reply:
Thx, added types of the output of per partition utility results